Jenkinsfile

```
@@ -0,0 +1,75 @@
+#! groovy
+@Library('corda-shared-build-pipeline-steps@5.0.1') _
+
+import com.r3.build.enums.BuildEnvironment
+import com.r3.build.utils.GitUtils
+
+int cpus = 1
+BuildEnvironment buildEnvironment = BuildEnvironment.AMD64_LINUX
+
+GitUtils gitUtils = new GitUtils(this)
+
+pipeline {
+    agent {
+        kubernetes {
+            cloud "eks-e2e"
+            yaml kubernetesBuildAgentYaml('build', buildEnvironment, cpus)
+            idleMinutes 15
+            podRetention always()
+            nodeSelector ([
+                    "kubernetes.io/arch=${buildEnvironment.buildArchitecture.toString().toLowerCase()}",
+                    "kubernetes.io/os=${buildEnvironment.buildOperatingSystem.toString().toLowerCase()}",
+                    "role=jenkins-agent"
+            ].join(','))
+            label ([
+                    "kubernetes-build-agent",
+                    "${cpus}cpus",
+                    "${buildEnvironment.buildArchitecture.toString().toLowerCase()}",
+                    "${buildEnvironment.buildOperatingSystem.toString().toLowerCase()}"
+            ].join('-'))
+            showRawYaml false
+            defaultContainer 'build'
+        }
+    }
+
+  parameters {
+    booleanParam defaultValue: (gitUtils.isReleaseBranch() || gitUtils.isReleaseTag(env.TAG_NAME)), description: 'Publish artifacts to Artifactory?', name: 'DO_PUBLISH'
+  }
+
+    environment {
+        GRADLE_PERFORMANCE_TUNING = '--parallel --build-cache'
+        GRADLE_USER_HOME = "/host_tmp/gradle"
+        ARTIFACTORY_CREDENTIALS = credentials('artifactory-credentials')
+        CORDA_ARTIFACTORY_PASSWORD = "${env.ARTIFACTORY_CREDENTIALS_PSW}"
+        CORDA_ARTIFACTORY_USERNAME = "${env.ARTIFACTORY_CREDENTIALS_USR}"
+        CORDA_ARTIFACTORY_REPOKEY = "${isRelease ? 'corda-dependencies' : 'corda-dependencies-dev'}"
+    }
+
+    stages {
+        stage('Prep') {
+            steps {
+                // Ensure the dir exists in the mount from the host
+                sh 'mkdir -p ${GRADLE_USER_HOME}'
+            }
+        }
+        stage('Compile') {
+            steps {
+                sh "./gradlew clean assemble"
+            }
+        }
+        stage('Test') {
+            steps {
+                sh "./gradlew test"
+            }
+        }
+        stage('Publish') {
+            when {
+                expression { params.DO_PUBLISH }
+                beforeAgent true
+            }
+            steps {
+                sh "./gradlew artifactoryPublish"
+            }
+        }
+    }
+}
```

```suggestion
}

```

new line at end of file will remove that GH warning/error